process/process.go

```diff
@@ -230,20 +235,61 @@ func (sp *systemProcess) GetMappings() ([]Mapping, uint32, error) {
 	defer mapsFile.Close()
 
 	mappings, numParseErrors, err := parseMappings(mapsFile)
-	if err == nil {
-		fileToMapping := make(map[string]*Mapping, len(mappings))
-		for idx := range mappings {
-			m := &mappings[idx]
-			if m.Inode == 0 {
-				// Ignore mappings that are invalid,
-				// non-existent or are special pseudo-files.
-				continue
-			}
-			fileToMapping[m.Path] = m
+	if err != nil {
```

Comment:
I think here we should also return early if `err` is `nil` and `numParseErrors` is non-zero. Or perhaps even better, `parseMappings` could return an `err` if it failed to find usable mappings (but it managed to read data). The idea is basically to distinguish here if mappings is empty or all lines were non-parseable.

Reply:
Doesn't the `len(mappings) == 0` check that follows cover this case? 

If `err == nil` and `len(mappings) != 0` then we simply continue and process the mappings. If `err == nil` and `len(mappings) == 0` then we continue and try mappings from another thread.